perception/mil_vision/ros_tools/colorspaces.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+from __future__ import division, print_function
+import numpy as np
+import cv2
+import argparse
+
+
+def viz_colorspaces(rgb_image, viz_scale):
+
+    '''
+    Returns image that shows the channels of the BGR, HSV, and LAB colorspace
+    representations of a given image
+    '''
+
+    w = int(rgb_image.shape[1] * viz_scale)
+    h = int(rgb_image.shape[0] * viz_scale)
+    viz = np.zeros((h * 3, w * 4, 3), dtype=np.uint8)
+
+    rgb_small = cv2.resize(rgb_image, dsize=(w, h))
+    hsv = cv2.cvtColor(rgb_small, cv2.COLOR_BGR2HSV)
+    lab = cv2.cvtColor(rgb_small, cv2.COLOR_BGR2LAB)
+
+    viz[h:h * 2, :w] = rgb_small
+    viz[:h, w:w * 2, :], viz[:h, w * 2:w * 3, :], viz[:h, w * 3:w * 4, :] = \
+        map(lambda x: np.repeat(x[:, :, np.newaxis], 3, axis=2), cv2.split(rgb_small))
+    viz[h:h * 2, w:w * 2, :], viz[h:h * 2, w * 2:w * 3, :], viz[h:h * 2, w * 3:w * 4, :] = \
+        map(lambda x: np.repeat(x[:, :, np.newaxis], 3, axis=2), cv2.split(hsv))
+    viz[h * 2:, w:w * 2, :], viz[h * 2:, w * 2:w * 3, :], viz[h * 2:, w * 3:w * 4, :] = \
+        map(lambda x: np.repeat(x[:, :, np.newaxis], 3, axis=2), cv2.split(lab))
+
+    font = cv2.FONT_HERSHEY_PLAIN
+    color = (255, 255, 255)
+    cv2.putText(viz, 'B', (1 * w, int(h * 0.15)), font, viz_scale * 5, (255, 0, 0), 2)
+    cv2.putText(viz, 'G', (2 * w, int(h * 0.15)), font, viz_scale * 5, (0, 255, 0), 2)
+    cv2.putText(viz, 'R', (3 * w, int(h * 0.15)), font, viz_scale * 5, (0, 0, 255), 2)
+    cv2.putText(viz, 'H', (1 * w, int(h * 1.15)), font, viz_scale * 5, color, 2)
+    cv2.putText(viz, 'S', (2 * w, int(h * 1.15)), font, viz_scale * 5, color, 2)
+    cv2.putText(viz, 'V', (3 * w, int(h * 1.15)), font, viz_scale * 5, color, 2)
+    cv2.putText(viz, 'L', (1 * w, int(h * 2.15)), font, viz_scale * 5, color, 2)
+    cv2.putText(viz, 'A', (2 * w, int(h * 2.15)), font, viz_scale * 5, color, 2)
+    cv2.putText(viz, 'B', (3 * w, int(h * 2.15)), font, viz_scale * 5, color, 2)
+
+    # TODO: overlay corresponding histogram over each of the images
+
+    return viz
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-c', '--use_cam', help='Use webcam', action='store_true')
+    parser.add_argument('-v', '--use_video', help='Use video', type=str)
+    parser.add_argument('-r', '--use_ros_topic', help='Use ROS topic', type=str)
+    parser.add_argument('-s', '--scale', help='Scale for image processing', type=float, default=0.5)
+    args = parser.parse_args()
+
+    if args.use_cam or args.use_video:
+        cap = None
+        if args.use_cam:
+            cap = cv2.VideoCapture(0)
+        else:
+            cap = cv2.VideoCapture(args.use_video)
+
+        while(True):
+            success, rgb_orig = cap.read()
+
+            if not success:
+                print("Unable to get frame")
+                break
+
+            cv2.imshow('Colorspaces - Press any key to exit', viz_colorspaces(rgb_orig, args.scale))
```

Comment:
Just a suggestion, but it may be a good idea to have an option to specify the scale of the window in pixels, or to just fit the window to the screen. I can see this getting annoying with images of different sizes.